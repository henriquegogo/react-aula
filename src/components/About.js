import React, { Component } from 'react';

export default class About extends Component {
  render() {
    return (
      <main>
        <h1>Aplicação massa</h1>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu leo dui, vitae tincidunt tellus. Donec quam quam, pretium ut posuere sed, interdum in nisl. Nullam pulvinar fermentum nunc, a euismod leo malesuada et. Nunc fermentum diam tempor nisi condimentum auctor. Curabitur accumsan elit vitae nisi tristique in tempus dolor accumsan. Sed tincidunt rutrum pharetra. Donec faucibus, arcu eu auctor egestas, ligula sem congue ante, at dapibus diam sapien sollicitudin neque. Nullam non sapien rutrum eros tempor aliquam. Nulla molestie congue odio id aliquam. Nam eget velit at nisl dignissim luctus. Etiam iaculis ipsum id eros auctor ut interdum sapien viverra.</p>
      </main>
    );
  }
}
