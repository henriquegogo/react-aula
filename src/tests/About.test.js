import React from 'react';
import { shallow } from 'enzyme';

import About from '../components/About';

describe('About', () => {
  let component;

  beforeEach(() => {
    component = shallow(<About />);
  })

  it('deve ser visível', () => {
    expect(component).toMatchSnapshot();
  })

  it('tem título', () => {
    expect(component.contains('Aplicação massa')).toEqual(true);
  })

  it('título se altera', () => {
    expect(component.contains(<h1>Aplicação massa</h1>)).toEqual(true);
    expect(component.state('foiClicado')).toEqual(false);

    const botao = component.find('button');
    botao.simulate('click');

    expect(component.contains(<h1>Novo título</h1>)).toEqual(true);
    expect(component.state('foiClicado')).toEqual(true);
  })

  it('título se altera independente do componente', () => {
    expect(component.contains(<h1>Aplicação massa</h1>)).toEqual(true);

    const novotitulo = 'O que eu quiser';
    component.setState({ titulo: novotitulo });

    expect(component.contains(novotitulo)).toEqual(true);
  })
})
