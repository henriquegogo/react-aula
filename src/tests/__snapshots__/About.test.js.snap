// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About deve ser visível 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <About />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Aplicação massa
        </h1>,
        <button
          onClick={[Function]}
        >
          Altera título
        </button>,
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu leo dui, vitae tincidunt tellus. Donec quam quam, pretium ut posuere sed, interdum in nisl. Nullam pulvinar fermentum nunc, a euismod leo malesuada et. Nunc fermentum diam tempor nisi condimentum auctor. Curabitur accumsan elit vitae nisi tristique in tempus dolor accumsan. Sed tincidunt rutrum pharetra. Donec faucibus, arcu eu auctor egestas, ligula sem congue ante, at dapibus diam sapien sollicitudin neque. Nullam non sapien rutrum eros tempor aliquam. Nulla molestie congue odio id aliquam. Nam eget velit at nisl dignissim luctus. Etiam iaculis ipsum id eros auctor ut interdum sapien viverra.
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Aplicação massa",
        },
        "ref": null,
        "rendered": "Aplicação massa",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Altera título",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Altera título",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu leo dui, vitae tincidunt tellus. Donec quam quam, pretium ut posuere sed, interdum in nisl. Nullam pulvinar fermentum nunc, a euismod leo malesuada et. Nunc fermentum diam tempor nisi condimentum auctor. Curabitur accumsan elit vitae nisi tristique in tempus dolor accumsan. Sed tincidunt rutrum pharetra. Donec faucibus, arcu eu auctor egestas, ligula sem congue ante, at dapibus diam sapien sollicitudin neque. Nullam non sapien rutrum eros tempor aliquam. Nulla molestie congue odio id aliquam. Nam eget velit at nisl dignissim luctus. Etiam iaculis ipsum id eros auctor ut interdum sapien viverra.",
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu leo dui, vitae tincidunt tellus. Donec quam quam, pretium ut posuere sed, interdum in nisl. Nullam pulvinar fermentum nunc, a euismod leo malesuada et. Nunc fermentum diam tempor nisi condimentum auctor. Curabitur accumsan elit vitae nisi tristique in tempus dolor accumsan. Sed tincidunt rutrum pharetra. Donec faucibus, arcu eu auctor egestas, ligula sem congue ante, at dapibus diam sapien sollicitudin neque. Nullam non sapien rutrum eros tempor aliquam. Nulla molestie congue odio id aliquam. Nam eget velit at nisl dignissim luctus. Etiam iaculis ipsum id eros auctor ut interdum sapien viverra.",
        "type": "p",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Aplicação massa
          </h1>,
          <button
            onClick={[Function]}
          >
            Altera título
          </button>,
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu leo dui, vitae tincidunt tellus. Donec quam quam, pretium ut posuere sed, interdum in nisl. Nullam pulvinar fermentum nunc, a euismod leo malesuada et. Nunc fermentum diam tempor nisi condimentum auctor. Curabitur accumsan elit vitae nisi tristique in tempus dolor accumsan. Sed tincidunt rutrum pharetra. Donec faucibus, arcu eu auctor egestas, ligula sem congue ante, at dapibus diam sapien sollicitudin neque. Nullam non sapien rutrum eros tempor aliquam. Nulla molestie congue odio id aliquam. Nam eget velit at nisl dignissim luctus. Etiam iaculis ipsum id eros auctor ut interdum sapien viverra.
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Aplicação massa",
          },
          "ref": null,
          "rendered": "Aplicação massa",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Altera título",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Altera título",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu leo dui, vitae tincidunt tellus. Donec quam quam, pretium ut posuere sed, interdum in nisl. Nullam pulvinar fermentum nunc, a euismod leo malesuada et. Nunc fermentum diam tempor nisi condimentum auctor. Curabitur accumsan elit vitae nisi tristique in tempus dolor accumsan. Sed tincidunt rutrum pharetra. Donec faucibus, arcu eu auctor egestas, ligula sem congue ante, at dapibus diam sapien sollicitudin neque. Nullam non sapien rutrum eros tempor aliquam. Nulla molestie congue odio id aliquam. Nam eget velit at nisl dignissim luctus. Etiam iaculis ipsum id eros auctor ut interdum sapien viverra.",
          },
          "ref": null,
          "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent eu leo dui, vitae tincidunt tellus. Donec quam quam, pretium ut posuere sed, interdum in nisl. Nullam pulvinar fermentum nunc, a euismod leo malesuada et. Nunc fermentum diam tempor nisi condimentum auctor. Curabitur accumsan elit vitae nisi tristique in tempus dolor accumsan. Sed tincidunt rutrum pharetra. Donec faucibus, arcu eu auctor egestas, ligula sem congue ante, at dapibus diam sapien sollicitudin neque. Nullam non sapien rutrum eros tempor aliquam. Nulla molestie congue odio id aliquam. Nam eget velit at nisl dignissim luctus. Etiam iaculis ipsum id eros auctor ut interdum sapien viverra.",
          "type": "p",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
